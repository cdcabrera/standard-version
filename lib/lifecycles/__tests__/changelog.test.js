const fs = require('fs');
const changelog = require('../changelog');

const CHANGELOG = `
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [2.0.0](https://github.com/conventional-changelog/standard-version/compare/v1.1.0...v2.0.0) (2022-06-02)

### Features

* **deprecated:** add deprecation message ([#907](https://github.com/conventional-changelog/standard-version/issues/907)) ([XXXXXXX](https://github.com/conventional-changelog/standard-version/commit/XXXXXXX))

<a name="1.1.0"></a>
# [1.1.0](https://github.com/conventional-changelog/standard-version/compare/v1.0.0...v1.1.0) (2022-06-01)

### Bug Fixes

* corrected an issue ([#123](https://github.com/conventional-changelog/standard-version/issues/123)) ([XXXXXXX](https://github.com/conventional-changelog/standard-version/commit/XXXXXXX))
`;

describe('changelog', () => {
  const mockInfo = jest.fn();
  let spy;

  beforeEach(() => {
    spy = jest.spyOn(console, 'info').mockImplementation(mockInfo);
  });

  afterEach(() => {
    mockInfo.mockClear();
    spy.mockClear();
  });

  it('should skip setting up changelog', async () => {
    const changelogInvalid = await changelog({ skip: { changelog: true } });

    expect({
      changelogInvalid
    }).toMatchSnapshot('results');
    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should attempt to update changelog', async () => {
    const { path: filepath } = generateFixture(CHANGELOG, {
      filename: 'CHANGELOG_ATTEMPT.md'
    });

    await changelog({
      skip: {},
      infile: filepath
    });

    const changelogUpdate = fs.readFileSync(filepath, { encoding: 'utf8' });

    expect(changelogUpdate).toContain('undefined');
    expect({
      changelogUpdate
    }).toMatchSnapshot('results');
    expect(mockInfo.mock.calls.map(results => results.map(result => result.replace(__basedir, '')))).toMatchSnapshot(
      'console'
    );
  });

  it('should update changelog', async () => {
    const { path: filepath } = generateFixture(CHANGELOG, {
      filename: 'CHANGELOG_UPDATE.md'
    });

    await changelog(
      {
        skip: {},
        infile: filepath,
        header: 'lorem ipsum'
      },
      '3.0.0'
    );

    const changelogUpdate = fs.readFileSync(filepath, { encoding: 'utf8' });

    expect(changelogUpdate).toContain('lorem ipsum');
    expect(changelogUpdate).toContain('[3.0.0]');
    expect(mockInfo.mock.calls.map(results => results.map(result => result.replace(__basedir, '')))).toMatchSnapshot(
      'console'
    );
  });
});
