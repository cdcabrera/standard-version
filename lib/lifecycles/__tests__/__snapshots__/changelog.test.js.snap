// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changelog should attempt to update changelog: console 1`] = `
Array [
  Array [
    "<green>✔</green> outputting changes to <bold>/.fixtures/CHANGELOG_ATTEMPT.md</bold>",
  ],
]
`;

exports[`changelog should attempt to update changelog: results 1`] = `
Object {
  "changelogUpdate": "undefined
## [2.0.0](https://github.com/conventional-changelog/standard-version/compare/v1.1.0...v2.0.0) (2022-06-02)

### Features

* **deprecated:** add deprecation message ([#907](https://github.com/conventional-changelog/standard-version/issues/907)) ([XXXXXXX](https://github.com/conventional-changelog/standard-version/commit/XXXXXXX))

<a name=\\"1.1.0\\"></a>
# [1.1.0](https://github.com/conventional-changelog/standard-version/compare/v1.0.0...v1.1.0) (2022-06-01)

### Bug Fixes

* corrected an issue ([#123](https://github.com/conventional-changelog/standard-version/issues/123)) ([XXXXXXX](https://github.com/conventional-changelog/standard-version/commit/XXXXXXX))
",
}
`;

exports[`changelog should skip setting up changelog: console 1`] = `Array []`;

exports[`changelog should skip setting up changelog: results 1`] = `
Object {
  "changelogInvalid": undefined,
}
`;

exports[`changelog should update changelog: console 1`] = `
Array [
  Array [
    "<green>✔</green> outputting changes to <bold>/.fixtures/CHANGELOG_UPDATE.md</bold>",
  ],
]
`;
